### PR TITLE
Bump version to 0.4, add preliminary and incomplete changelog for opt…

### PR DESCRIPTION
**Pull request by phadej**
…ics-core

I need a version bump, so I can prepare for a new release. (Getting CPP right).

I added some changelog, what I could tell from API diff. There are few `@since` annotations. (Probably too few).

I think @arybczak is the best one to write about changes related to `LabelOptic`

### CODE REVIEW

**Comment by arybczak** [comment]
I think migration guide should be separate from the changelog.

**Comment by phadej** [comment]
If you want to separate it, feel free. Remember, Hackage shows only Readme and a link to changelog, so if we separate migration guide out of changelog, we should have a link there.

**Comment by phadej** [comment]
I have split the migration notes into separate file.

File: template-haskell-optics/CHANGELOG.md [hunk 1]
```diff
@@ -1,2 +1,5 @@
+# template-haskell-optics-0.2
+* TBW: No DataPrism
```
Comment (arybczak) [inline template-haskell-optics/CHANGELOG.md:2]: `DataPrism` wasn't exported, so it probably doesn't matter.

Comment (phadej) [reply template-haskell-optics/CHANGELOG.md:2]: I defer completion and correction of that changelog to you for later.

**Comment by phadej** [comment]
I think this is good enough. Further improvements can be done in the follow-ups.

**Pull request merged**